src/libraries/Delegator.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/libraries/DelegationTarget.sol';
+
+
+contract Delegator is DelegationTarget {
+
+    function Delegator(address controllerAddress, bytes32 keyVal) {
+        controller = Controller(controllerAddress);
+        controllerLookupName = keyVal;
+    }
+
+    // We currently only support a single non-array return value
+    // If we truly need to return more advanced sets of data we could follow
+    // the pattern demonstrated here: https://gist.github.com/Arachnid/4ca9da48d51e23e5cfe0f0e14dd6318f
```

Comment:
Totally agree. I thought the idea was neat although the implementation was too potentially fraught with accidents unless we automated it.